knimin/lib/tests/test_data_access.py

```diff
@@ -1171,6 +1171,137 @@ def test_delete_dna_plate(self):
             db.delete_dna_plate(0)
         self.assertEqual(ctx.exception.message, "DNA plate 0 does not exist")
 
+    def test_get_dna_plate_list(self):
+        self.assertEqual(db.get_dna_plate_list(), [])
+
+        # Create a study
+        db.create_study(9999, title='LabAdmin test project', alias='LTP',
+                        jira_id='KL9999')
+        self._clean_up_funcs.append(partial(db.delete_study, 9999))
+        # Create a sample plate
+        pt = db.get_plate_types()[0]
+        plate_id = db.create_sample_plate('Test plate', pt['id'],
+                                          'test', [9999])
+        self._clean_up_funcs.insert(
+            0, partial(db.delete_sample_plate, plate_id))
+
+        # Create a dna plate
+        exp_robot = db.get_property_options("extraction_robot")[0]
+        exp_kit = db.get_property_options("extraction_kit_lot")[0]
+        exp_tool = db.get_property_options("extraction_tool")[0]
+        obs = db.extract_sample_plates(
+            [plate_id], 'test', exp_robot['name'], exp_kit['name'],
+            exp_tool['name'])
+        self._clean_up_funcs.insert(0, partial(db.delete_dna_plate, obs[0]))
+
+        exp = [{'id': obs[0], 'name': 'Test plate',
+                'date': datetime.date.today()}]
+        self.assertEqual(db.get_dna_plate_list(), exp)
+
+    def test_get_targeted_primer_plates(self):
+        exp = [{'id': 1, 'name': 'Primer plate 1', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 2, 'name': 'Primer plate 2', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 3, 'name': 'Primer plate 3', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 4, 'name': 'Primer plate 4', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 5, 'name': 'Primer plate 5', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 6, 'name': 'Primer plate 6', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 7, 'name': 'Primer plate 7', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'},
+               {'id': 8, 'name': 'Primer plate 8', 'notes': None,
+                'linker_primer_sequence': 'GTGTGCCAGCMGCCGCGGTAA',
+                'target_gene_region': '16S V4'}]
```

Comment:
I think the column used in qiita preps is "target_gene" and "target_subfragment". Would it be worth following that?